hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusSubmitMessageHandler.java

```diff
@@ -143,26 +143,22 @@ public void handle(@NonNull final HandleContext handleContext) {
      * @param config the {@link ConsensusConfig}
      * @param topic the topic to which the message is being submitted
      */
-    private void validateTransaction(
-            final TransactionBody txn, final ConsensusConfig config, final Optional<Topic> topic) {
+    private void validateTransaction(final TransactionBody txn, final ConsensusConfig config, final Topic topic) {
         final var txnId = txn.transactionID();
         final var payer = txn.transactionIDOrElse(TransactionID.DEFAULT).accountIDOrElse(AccountID.DEFAULT);
         final var op = txn.consensusSubmitMessageOrThrow();
 
         /* Check if the message submitted is empty */
         // Question do we need this check ?
         final var msgLen = op.message().length();
-        if (msgLen == 0) {
-            throw new HandleException(INVALID_TOPIC_MESSAGE);
-        }
 
         /* Check if the message submitted is greater than acceptable size */
         if (msgLen > config.messageMaxBytesAllowed()) {
             throw new HandleException(MESSAGE_SIZE_TOO_LARGE);
         }
 
         /* Check if the topic exists */
-        if (topic.isEmpty()) {
+        if (topic == null) {
```

Comment:
Should we also check for the default topic here?

Reply:
This is the topic we get from XXXStore. I don't think we have default value returned from store. If not found in the store, just return null.